crates/relayer-cli/src/commands/tx/channel.rs

```diff
@@ -131,8 +145,96 @@ impl Runnable for TxChanOpenInitCmd {
+
+        // Check if connection IDs were provided via --connection-hops, indicating a multi-hop channel
+        if let Some(connnection_ids) = &self.conn_hop_ids {
+            // Retrieve information for each of the remaining hops until the other end of the channel is reached
+            for connection_id in connnection_ids.as_slice().iter().rev() {
+                // Retrieve the ChainId of the chain referenced by the previous connection hop
+                let chain_id = &assembled_hops.last().unwrap().reference_chain_id;
+
+                // Spawn a handle for the chain referenced in the previous hop
+                let chain_handle = match spawn_chain_runtime(&config, chain_id) {
+                    Ok(handle) => handle,
+                    Err(e) => Output::error(e).exit(),
+                };
+
+                // Retrieve the connection associated with the next hop
+                let hop_connection = match chain_handle.query_connection(
+                    QueryConnectionRequest {
+                        connection_id: connection_id.clone(),
+                        height: QueryHeight::Latest,
+                    },
+                    IncludeProof::No,
+                ) {
+                    Ok((connection, _)) => connection,
+                    Err(e) => Output::error(e).exit(),
+                };
+
+                // Retrieve the state of the client underlying the hop connection
+                let hop_conn_client_state = match chain_handle.query_client_state(
+                    QueryClientStateRequest {
+                        client_id: hop_connection.client_id().clone(),
+                        height: QueryHeight::Latest,
+                    },
+                    IncludeProof::No,
+                ) {
+                    Ok((client_state, _)) => client_state,
+                    Err(e) => Output::error(e).exit(),
+                };
+
+                assembled_hops.push(ConnectionHop {
+                    connection: IdentifiedConnectionEnd::new(
+                        connection_id.clone(),
+                        hop_connection.clone(),
+                    ),
+                    reference_chain_id: hop_conn_client_state.chain_id().clone(),
+                });
+            }
+        }
+
+        // FIXME: For now, pass Some(_) to connection_hops if there are multiple hops and None if there is a single one.
+        // This allows us to keep using existing structs as they are defined (with the single `connection_id` field) while also including
+        // the new `connection_hops` field. When multiple hops are present, pass Some(_) to connection_hops and use that.
+        // When a single hop is present, pass None to connection_hops and use the connection_id stored in `ChannelSide`.
+        let connection_hops = match assembled_hops.len() {
+            0 => Output::error("At least one connection hop is required for opening a channel.")
+                .exit(),
+            1 => None,
```

Comment:
Why is this required? We could still use the same structure but always use the `connection_hops` when we create the handshake message.
(would get rid of `.unwrap_or_else(|| vec![self.dst_connection_id().clone()]),` further down the stack (`crates/relayer/src/channel.rs::build_chan_open_init()`)

Reply:
Yes we absolutely plan on doing this at a later stage, but for now we wanted to keep as much of the existing code as is to the extent possible.